polar-core/src/vm.rs

```diff
@@ -719,7 +719,7 @@ impl PolarVirtualMachine {
     /// The exception is for lists, so that we can correctly handle rest variables.
     /// We also support cycle detection, in which case we return the original term.
     fn deref(&self, term: &Term) -> Term {
```

Comment:
I don't think we need a `deref` function and a `deep_deref` function if they both do the same thing. What do you think?

Reply:
No, of course I agree. I'll do what you suggested and try to replace `deref` with `variable_state` where possible though.

Reply:
(there turned out to be only one such case)